practice_app/api/models.py

```diff
@@ -1,3 +1,57 @@
 from django.db import models
+from django.contrib.auth.models import User
+
+import json
 
 # Create your models here.
+
+class Paper(models.Model):
+    paper_id = models.AutoField(primary_key=True)
+    third_party_id = models.CharField(max_length=100)
+    source = models.CharField(max_length=50)
+    abstract = models.TextField(max_length=5000, blank=True, null=True)
+    year = models.IntegerField(blank=True, null=True)
+    url = models.URLField()
+    authors = models.TextField(max_length=500, blank=True, null=True)
+    title = models.CharField(max_length=100)
+    like_count = models.IntegerField()
+
+    def set_authors(self, authors):
+        self.authors = json.dumps(authors)
+
+    def get_authors(self):
+        return json.loads(self.authors) if self.authors else []
+
+class Comment(models.Model):
+    # comment_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    paper =  models.ForeignKey(Paper, on_delete=models.CASCADE)
+    text = models.TextField(max_length=500)
+
+class Like(models.Model):
+    # like_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    paper = models.ForeignKey(Paper, on_delete=models.CASCADE)
+
+
+class UserInterest(models.Model):
+    # interest_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    interest = models.CharField(max_length=50, blank=True, null=True)
+
+class PaperList(models.Model):
+    #list_id = models.AutoField(primary_key=True)
+    list_title = models.CharField(max_length=50, default='Paper List')
+    paper = models.ManyToManyField(Paper, blank=True, null=True)
+    owner = models.ForeignKey(User, related_name='owner', on_delete=models.CASCADE)
+    saver = models.ManyToManyField(User, related_name='savers', blank=True, null=True)
+
+class FollowRequest(models.Model):
```

Comment:
Can a user send a follow request to a user more than once ?

Reply:
No, the status field in FollowRequest model will be used to reflect the current status of the request. Also I think this case can be handled in the implementation of the relevant POST method using the status field.

Reply:
Then, the follower and the receiver should not be composit key?

Reply:
You are right about the uniqueness of the sender and receiver fields. I will add uniqueness constraint for them as well.